pkg/cluster/internal/create/actions/createworker/utils.go

```diff
@@ -107,99 +65,152 @@ func generateB64Credentials(access_key string, secret_key string, region string)
+}
 
-	descriptorRAW, err := os.ReadFile("./" + descriptorName)
-	if err != nil {
-		return err
-	}
+func checkAWSCreds(awsCreds cluster.AWSCredentials) bool {
+	awsEmptyCreds := cluster.AWSCredentials{}
+	return awsCreds == awsEmptyCreds
+}
 
-	descriptorMap := map[string]interface{}{}
-	viper.SetConfigName(descriptorName)
-	currentDir, err := currentdir()
-	if err != nil {
-		fmt.Println(err)
-		return err
-	}
-	viper.AddConfigPath(currentDir)
+func checkGHToken(github_token string) bool {
+	return github_token == ""
+}
+
+func ensureSecretsFile(descriptorFile cluster.DescriptorFile, vaultPassword string) error {
+	edited := false
 
-	err = yaml.Unmarshal(descriptorRAW, &descriptorMap)
+	awsCredentials, github_token, err := getCredentials(descriptorFile, vaultPassword)
 	if err != nil {
 		return err
 	}
 
-	if descriptorMap["aws"] != nil || descriptorMap["github_token"] != nil {
-		deleteKey("aws", descriptorMap)
-		deleteKey("github_token", descriptorMap)
-
-		d, err := yaml.Marshal(&descriptorMap)
+	_, err = os.Stat(secretPath)
+	if err != nil {
+		secret := Secret{awsCredentials, github_token}
+		secretFile := SecretsFile{secret}
+		secretRaw, err := yaml.Marshal(secretFile)
+		secretMap := map[string]map[string]interface{}{}
+		err = yaml.Unmarshal([]byte(secretRaw), &secretMap)
 		if err != nil {
-			fmt.Println("error: %v", err)
 			return err
 		}
 
-		// write to file
-		f, err := os.Create(currentDir + descriptorName)
+		err = encryptSecret(secretMap, vaultPassword)
 		if err != nil {
-			fmt.Println(err)
-			return nil
+			return err
 		}
+		return nil
+	}
+	// En caso de que exista
 
-		err = ioutil.WriteFile(descriptorName, d, 0755)
+	secretRaw, err := decryptFile(secretPath, vaultPassword)
+	if err != nil {
+		return err
+	}
+	secretMap := map[string]map[string]interface{}{}
+	err = yaml.Unmarshal([]byte(secretRaw), &secretMap)
+	if err != nil {
+		return err
+	}
+
+	if secretMap["secrets"]["aws"] == nil {
+		edited = true
+		secretMap["secrets"]["aws"] = awsCredentials
+	}
+	if secretMap["secrets"]["github_token"] == nil {
+		edited = true
+		secretMap["secrets"]["github_token"] = github_token
+	}
+	if edited {
+		err = encryptSecret(secretMap, vaultPassword)
 		if err != nil {
-			fmt.Println("error: %v", err)
 			return err
 		}
+		return nil
+	}
+	return nil
+}
 
-		f.Close()
+func rewriteDescriptorFile(descriptorName string) error {
 
+	descriptorRAW, err := os.ReadFile("./" + descriptorName)
+
+	if err != nil {
+		return err
 	}
 
-	return nil
+	var data yaml.Node
+	err = yaml.Unmarshal(descriptorRAW, &data)
+	if err != nil {
+		return err
+	}
 
-}
+	nodes := removeNode(data.Content, "aws")
+	nodes = removeNode(data.Content, "github_token")
+
+	b, err := yaml.Marshal(nodes[0])
 
-func deleteKey(key string, descriptorMap map[string]interface{}) {
-	value := descriptorMap[key]
-	if value != nil {
-		delete(descriptorMap, key)
+	fmt.Println(string(b))
+
+	err = ioutil.WriteFile(descriptorName, b, 0755)
```

Comment:
```suggestion
	err = ioutil.WriteFile(descriptorName, b, 0644)
```
it shouldn't be executable